### PR TITLE
Replace HiWayEnv with "GymBaseEnv"

### PR DESCRIPTION
**Pull request by liu-allan**
#1279 

Not sure how to check if a combination is invalid here:
- details should be logged if a scenario&agent-interface combination is invalid to be able to start an episode

### CODE REVIEW

File: smarts/env/wrappers/single_agent.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def step(self, action: Any) -> Tuple[Any, float, bool, Any]:
         return (
             obs[self._agent_id],
             reward[self._agent_id],
-            done[self._agent_id],
+            done,
```
Comment (Adaickalavan) [inline smarts/env/wrappers/single_agent.py:61]: I think this should not be changed, as it is intended to wrap a multi-agent env to become compatible with a single-agent gym specs.
```suggestion
            done[self._agent_id],
```

File: smarts/core/utils/episodes.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def record_step(self, observations=None, rewards=None, dones=None, infos=None):
                 observations, rewards, dones, infos
             )
 
-        if dones.get("__all__", False) and infos is not None:
+        if dones and infos is not None:
```
Comment (Adaickalavan) [inline smarts/core/utils/episodes.py:77]: I think `dones` here is a `dict`. Why was this changed?

Comment (liu-allan) [reply smarts/core/utils/episodes.py:77]: From the issue #1279 one of the requests was that `done` be related to the environment being done rather than for individual agents. I have changed `done` to represent `dones["__all__"]` instead to reflect the request. See https://github.com/huawei-noah/SMARTS/pull/1374/files#diff-7f6d3d7e26af4d9bdf221b3951430a399520fd204bdf830e475813052c2c1d28R333

File: smarts/env/wrappers/single_agent.py (hunk 1 above)
Comment (liu-allan) [reply smarts/env/wrappers/single_agent.py:61]: See other comment

File: smarts/core/utils/episodes.py (hunk 2 above)
Comment (Gamenot) [reply smarts/core/utils/episodes.py:77]: This is correct for `Gym` specific environments. When we get around to `RLlib` style environments that may be different.

**Comment by Adaickalavan** [comment]
Closed due to becoming obsolete.
